spyder/plugins/toolbar/container.py

```diff
@@ -0,0 +1,322 @@
+from spyder.api.translations import get_translation
+from spyder.api.widgets import PluginMainContainer
+from spyder.api.widgets.toolbars import ApplicationToolBar
+from spyder.plugins.toolbar.api import ApplicationToolBars
+
+# Localization
+_ = get_translation('spyder')
+
+
+# --- Constants
+# ------------------------------------------------------------------------
+
+class ToolBarMenus:
+    ToolBarsMenu = "toolbars_menu"
+
+
+class ToolBarsMenuSections:
+    Main = "main_section"
+    Secondary = "secondary_section"
+
+
+class ToolBarActions:
+    ShowToolBars = "show_toolbars_actions"
+
+
+class ToolBarContainer(PluginMainContainer):
+    DEFAULT_OPTIONS = {
+        'last_visible_toolbars': [],
+        'toolbars_visible': True,
+    }
+
+    def __init__(self, name, plugin, parent=None, options=DEFAULT_OPTIONS):
+        super().__init__(name, plugin, parent=parent, options=options)
+
+        self._APPLICATION_TOOLBARS = OrderedDict()
+        self._ADDED_TOOLBARS = OrderedDict()
+        self._toolbarslist = []
+        self._visible_toolbars = []
+
+    # --- Private Methods
+    # ------------------------------------------------------------------------
+    def _save_visible_toolbars(self):
+        """Saves the name of the visible toolbars in the options."""
+        toolbars = []
+        for toolbar in self._visible_toolbars:
+            toolbars.append(toolbar.objectName())
+
+        # TODO: Move this to a different section since there is no concept
+        # of a MainPlugin (Unless it is created)
+        self.set_option('last_visible_toolbars', toolbars)
+
+    def _get_visible_toolbars(self):
+        """Collects the visible toolbars."""
+        toolbars = []
+        for toolbar in self._toolbarslist:
+            if toolbar.toggleViewAction().isChecked():
+                toolbars.append(toolbar)
+
+        self._visible_toolbars = toolbars
+
+    @Slot()
+    def _show_toolbars(self):
+        """Show/Hides toolbars."""
+        value = not self.get_option("toolbars_visible")
+        # TODO: Move this to a different section since there is no concept
+        # of a MainPlugin (Unless it is created)
+        self.set_option("toolbars_visible", value)
+        if value:
+            self._save_visible_toolbars()
+        else:
+            self._get_visible_toolbars()
+
+        for toolbar in self._visible_toolbars:
+            toolbar.toggleViewAction().setChecked(value)
+            toolbar.setVisible(value)
+
+        self.update_actions()
+
+    # --- PluginMainContainer API
+    # ------------------------------------------------------------------------
+    def setup(self, options=DEFAULT_OPTIONS):
+        self.show_toolbars_action = self.create_action(
+            ToolBarActions.ShowToolBars,
+            text=_("Show toolbars"),
+            triggered=self._show_toolbars,
+            context=Qt.ApplicationShortcut,
+            shortcut_context="_",
+            register_shortcut=True

```

Comment:
Did this action have an associated shortcut before? If not, please remember what we agreed on, i.e. not adding new shortcuts if they are not present in `4.x`.

Reply:
>Did this action have an associated shortcut before? 

Yes.

<img width="912" alt="Screen Shot 2020-08-31 at 20 05 56" src="https://user-images.githubusercontent.com/3627835/91783300-61cb0080-ebc5-11ea-8a80-e246d7744a2a.png">